File: LeetCode/Medium/Maximum Number Of Vowels In A Substring Of Given Length/Java/maxVowels.java
class Solution {
    public int maxVowels(String s, int k) {
        int cnt = 0, max = 0;
        int i = 0, j = 0;
        while(j < s.length()) {
            char c = s.charAt(j);
            if (j-i == k) {
                max = Math.max(max, cnt);
                if (isVowel(s.charAt(i))) cnt--;
                i++;
            }
            if (isVowel(s.charAt(j))) {
                cnt++;
            }
            j++;
        }
        max = Math.max(max, cnt);
        return max;
    }

    private boolean isVowel(char c) {
        return c == 'a' ||
            c == 'e' ||
            c == 'i' ||
            c == 'o' ||
            c == 'u';
    }
}

File: LeetCode/Medium/Maximum Number Of Vowels In A Substring Of Given Length/C++/max_vowels.cc
class Solution {
public:
    int maxVowels(string s, int k) {
        int i=0, max_vowel=0, curr=0;
        int j=0;
        while(j < s.length()) {
            if (j-i== k) {
                max_vowel = max(max_vowel, curr);
                if (is_vowel(s[i])) curr--;
                i++;
            }
            if (is_vowel(s[j])) curr++;
            j++;
        }
        max_vowel = max(max_vowel, curr);
        return max_vowel;
    }

    bool is_vowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
};
